rotkehlchen/chain/ethereum/manager.py

```diff
@@ -124,22 +124,26 @@ def _query_web3_get_logs(
         # is infura can throw an error here which we can only parse by catching the  exception
         try:
             new_events_web3: List[Dict[str, Any]] = [dict(x) for x in web3.eth.get_logs(filter_args)]  # noqa: E501
-        except ValueError as e:
-            try:
-                decoded_error = json.loads(str(e).replace("'", '"'))
-            except json.JSONDecodeError:
-                # reraise the value error if the error is not json
-                raise e from None
+        except (ValueError, KeyError) as e:
+            if isinstance(e, ValueError):
+                try:
+                    decoded_error = json.loads(str(e).replace("'", '"'))
+                except json.JSONDecodeError:
+                    # reraise the value error if the error is not json
+                    raise e from None
```

Comment:
I didn't know about this. Is `None` what we want here? As per https://www.python.org/dev/peps/pep-0409/#id10 I understand that the context is lost?

Reply:
We just reraise `e` essentially. The original error. So no context is lost. And I think the reason to do it like this is probably to satisfy a linter if memory serves ... even though I am not 100% sure.